apps/redirect/server/Dockerfile

```
@@ -0,0 +1,16 @@
+# Multi-stage build for the redirect tier (redirectd).
+# Produces a small static binary on a distroless base.
+FROM golang:1.26-alpine AS build
+WORKDIR /src
+# Cache deps first.
+COPY go.mod go.sum ./
+RUN go mod download
+COPY . .
+RUN CGO_ENABLED=0 GOOS=linux go build -trimpath -ldflags="-s -w" -o /out/redirectd ./cmd/redirectd
+
+FROM gcr.io/distroless/static-debian12
```

Hadolint (2.14.0)

[warning] Always tag the version of an image explicitly

(DL3006)